tiledb/common/random/prng.cc

```diff
@@ -33,16 +33,77 @@
 #include "tiledb/common/random/prng.h"
 
 namespace tiledb::common {
+/**
+ * 64-bit mersenne twister engine for random number generation.
+ *
+ * This definition is duplicated to avoid having it defined as `public` in
+ * `class PRNG`.
+ */
+using prng_type = std::mt19937_64;
+
+/**
+ * Implementation of the random seed.
+ *
+ * This is a class template in order to use `if constexpr`.
+ *
+ * @tparam return_size_type The type of the seed to be returned
+ */
+template <class return_size_type>
+return_size_type random_seed() {
+  static constexpr size_t rng_size = sizeof(std::random_device::result_type);
+  static constexpr size_t ret_size = sizeof(return_size_type);
+  std::random_device rng{};
+  /*
+   * We will need 64 bits to fully seed the PRNG (`ret_size`). We support cases
+   * where the result size of the RNG is 64 or 32 bits (`rng_size`).
```

Comment:
Just a heads up, but the possible seed state is much larger than 64 bits. I was poking at trying to get tiledb_unit to compile against an installed version of libtiledb without internal headers and realized I had to replace the PRNG since its not got an API. The implementation I came up with based on a few blogs and SO posts is at [1] if you're interested.

[1] https://github.com/TileDB-Inc/TileDB/compare/dev...pd/experiment/temporary-local-filesystem#diff-d047123bd694257fb7d998d0290f560532cb3e28961c49cc8d5d39773b682788R80-R117

Reply:
This code is using the abbreviated seeding constructor, the one that takes a `result_type`, here 64 bits. For the purpose of ensuring that seeds are distinct, it's plenty.

The long form seed, the one that takes a sequence type, initializes the full PRNG state. It's overkill for the immediate purpose. It's certainly useful for qualifying an implementation of the generator, but we don't need it. It might be useful for running a test from a known point in the PRNG state, but unfortunately, there's no way of getting such state out of the PRNG.

Reply:
Changed "fully" to "adequately"